### DISCUSSION TITLE
Play video if view in Program

### DISCUSSION THREAD

**Discussion by EliasDIPA** [WarmUpTill/SceneSwitcher#459] [Ideas]
Hello good day. It would help me a lot if there was a function to automatically play the video that is in the program. Without having to select each video source in the macro, for example:

if {{Media}}
{{All media sources on}} {{--select scene--}} state is **{{program view}}** and {{.....}} 0.00 {{.....}}
___
{{Media}}
{{play}}  **{{the video}} in {{program view}}** 

Note: in bold what should be added to Media

**Comment by WarmUpTill** [comment]
Am I understanding your request correctly that you basically want to do something like this:
* if all media sources on **current scene** are in state ...
* play all media sources on **current scene**

If not, can you please elaborate on what exactly you expect "state is {{program view}}" to represent?

**Comment by EliasDIPA** [comment]
<img width="898" alt="obs64_TqfNVXaZ6P" src="https://user-images.githubusercontent.com/104694558/167268783-57ea8ca6-5426-4436-9057-5b1b14e1a1a4.png">

When the plugin notices that there is a video in the program view, it plays that video, and that happens with any video you add to OBS, without the need to add the video to the macro.
If the macro detects that there is a video in Program, it plays the video, and when it does not detect it, it pauses the video.
That is my suggestion.
